src/test/regress/expected/validate_constraint.out

```diff
@@ -0,0 +1,256 @@
+ create_reference_table 
+------------------------
+ 
+(1 row)
+
+CREATE TABLE constrained_table (
+  id                 int,
+  constrained_column int,
+  ref_id             int
+);
+SELECT create_distributed_table('constrained_table', 'ref_id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- The two constraint types that are allowed to be NOT VALID
+BEGIN;
+ALTER TABLE constrained_table
+  ADD CONSTRAINT check_constraint CHECK (constrained_column > 100) NOT VALID;
+ALTER TABLE constrained_table
+  VALIDATE CONSTRAINT check_constraint;
+ROLLBACK;
+BEGIN;
+ALTER TABLE constrained_table
+  ADD CONSTRAINT fk_constraint FOREIGN KEY (ref_id) REFERENCES referenced_table (id) NOT VALID;
+ALTER TABLE constrained_table
+  VALIDATE CONSTRAINT fk_constraint;
+ROLLBACK;
+-- It is possible that some other table_constraint commands will support NOT VALID option in the future
+-- These should fail as Postgres does not yet allow these constraints to be NOT VALID
+-- If one of these queries are not failing on a future Postgres version, we will need to test corresponding VALIDATE CONSTRAINT queries
+BEGIN;
+ALTER TABLE constrained_table
+  ADD CONSTRAINT unique_constraint UNIQUE (constrained_column) NOT VALID;
+ERROR:  UNIQUE constraints cannot be marked NOT VALID
+ALTER TABLE constrained_table
+  VALIDATE CONSTRAINT unique_constraint;
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
+ROLLBACK;
+BEGIN;
+ALTER TABLE constrained_table
+  ADD CONSTRAINT pk_constraint PRIMARY KEY (id) NOT VALID;
+ERROR:  PRIMARY KEY constraints cannot be marked NOT VALID
+ALTER TABLE constrained_table
+  VALIDATE CONSTRAINT pk_constraint;
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
+ROLLBACK;
+BEGIN;
+ALTER TABLE constrained_table
+  ADD CONSTRAINT exclude_constraint EXCLUDE USING gist (id WITH =, constrained_column WITH <>) NOT VALID;
+ERROR:  EXCLUDE constraints cannot be marked NOT VALID
+ALTER TABLE constrained_table
+  VALIDATE CONSTRAINT exclude_constraint;
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
+ROLLBACK;
+-- drop table for next tests
+DROP TABLE constrained_table;
+DROP TABLE referenced_table;
+CREATE TABLE constrained_table (
+  id                 int,
+  constrained_column int
+);
+SELECT create_distributed_table('constrained_table', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+INSERT INTO constrained_table
+SELECT x, x
+from generate_series(1, 1000) as f (x);
+-- a constraint that can be validated
+ALTER TABLE constrained_table
+  ADD CONSTRAINT validatable_constraint CHECK (constrained_column < 10000) NOT VALID;
+ALTER TABLE constrained_table
+  VALIDATE CONSTRAINT validatable_constraint;
+-- a constraint that can not be validated
+ALTER TABLE constrained_table
+  ADD CONSTRAINT failing_constraint CHECK (constrained_column > 100) NOT VALID;
+ALTER TABLE constrained_table
```

Comment:
I remember suggesting adding such a test where the validation fails. But, now I'm a little bit skeptical to this. The reason is that, the output of this test might change randomly. In this test, probably the first shard is complaining about the violation. But, its possible that the same test errors out on the second shard and these error messages change.

So, given that you've `constraint_validations_in_workers ` view, I'd suggest to remove these test.

If you really want to have such a test, you should hide the shard names in the error messages. An example you can see is here: Define a function which absorbs the message as [1](https://github.com/citusdata/citus/pull/2475/files#diff-ddf4b11e6f488ae78926f36e2f9fb864R13) and call the command within that function [2](https://github.com/citusdata/citus/pull/2475/files#diff-ddf4b11e6f488ae78926f36e2f9fb864R316).

But, as I mentioned, given the view, we don't need it. I mostly pointed out for future reference.